include/coap2/resource.h

```diff
@@ -513,66 +475,13 @@ int coap_delete_observer(coap_resource_t *resource,
  */
 void coap_delete_observers(coap_context_t *context, coap_session_t *session);
 
-/**
- * Set whether a @p resource is observable.  If the resource is observable
- * and the client has set the COAP_OPTION_OBSERVE in a request packet, then
- * whenever the state of the resource changes (a call to
- * coap_resource_trigger_observe()), an Observer response will get sent.
- *
- * @param resource The CoAP resource to use.
- * @param mode     @c 1 if Observable is to be set, @c 0 otherwise.
- *
- */
-COAP_STATIC_INLINE void
-coap_resource_set_get_observable(coap_resource_t *resource, int mode) {
-  resource->observable = mode ? 1 : 0;
-}
-
-/**
- * Initiate the sending of an Observe packet for all observers of @p resource,
- * optionally matching @p query if not NULL
- *
- * @param resource The CoAP resource to use.
- * @param query    The Query to match against or NULL
- *
- * @return         @c 1 if the Observe has been triggered, @c 0 otherwise.
- */
-int
-coap_resource_notify_observers(coap_resource_t *resource,
-                               const coap_string_t *query);
-
-/**
- * Get the UriPath from a @p resource.
- *
- * @param resource The CoAP resource to check.
- *
- * @return         The UriPath if it exists or @c NULL otherwise.
- */
-COAP_STATIC_INLINE coap_str_const_t*
-coap_resource_get_uri_path(coap_resource_t *resource) {
-  if (resource)
-    return resource->uri_path;
-  return NULL;
-}
-
 /** @} */
 
-/**
- * Checks for all known resources, if they are dirty and notifies subscribed
- * observers.
- */
-void coap_check_notify(coap_context_t *context);
-
 coap_print_status_t coap_print_wellknown(coap_context_t *,
```

Comment:
I think this should be moved to `coap_resource_internal.h` as well.

Reply:
Moved, and had to include `coap_internal.h` in the appropriate test.